src/robot/reporting/xunitwriter.py

```diff
@@ -57,6 +60,14 @@ def _get_stats(self, statistics):
         )
 
     def end_suite(self, suite):
+        if suite.metadata or suite.doc:
+            self._writer.start('properties')
+            if suite.doc:
+                self._writer.element('property', attrs={'name': 'Suite Documentation', 'value': suite.doc})
+            if suite.metadata:
+                for n, v in suite.metadata.items():
+                    self._writer.element('property', attrs={'name': n, 'value': v})
+            self._writer.end('properties')
```

Comment:
No need for `if` here, it would be fine to just loop over possible metadata items. Using `name` and `value´ instead of `n` and `w` would be a bit better as well. We've generally used one character variable names only in very tight scopes such as in list comprehensions.